packages/laconia-event/src/apigateway/parseWebSocket.js

```diff
@@ -2,6 +2,6 @@ const getBody = require("./getBody");
 const tryParseJson = require("../tryParseJson");
 
 module.exports = event => {
-  const body = getBody(event);
+  const body = event.body && getBody(event);
```

Comment:
Maybe a better solution would be make that verification at getBody
But this could change behavior of other events Adapters.
@geoffdutton any observation?

Reply:
Yeah my initial reaction would to put it in `getBody`, but then that poses the question, is `parseWebSocket.js` necessary?

I'm not as familiar with the website socket events. So an event can come from Lambda where the `.body` is undefined?

Reply:
API Gateway websocket have at least 2 events:
- Connect (no body)
- Message (with body)
- Disconnect (??)

There are no well defined behavior if our user should set one handler for all these events or different handlers.

Reply:
Got it. I think what you have makes sense then considering a normal API Gateway event will always have a body. And then it makes sense to have a explicit `parseWebSocket` function as well because we might want to further parse out things specific to websockets. So I'd say you're good.